_source/_posts/2022-09-08-react-fastify-postgres.md

```diff
@@ -0,0 +1,897 @@
+---
+layout: blog_post
+title: ""
+author: gabi-dombrowski
+by: contractor
+communities: [javascript]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+
+In this tutorial, we'll set up an app to help us keep track of what NASA facilities we've visited and which ones we still want to visit.
+
+Our app will be a monorepo with Okta authentication using React for the frontend and Fastify for the backend. [Fastify](https://www.fastify.io/) is a highly performant web framework with low overhead that we'll connect to a Postgresql database. We'll also use [Lerna](https://lerna.js.org/) to manage the frontend and backend apps in a monorepo.
+
+**Prerequisites**
+
+*Create React App* currently requires Node >= 14.0.0 and npm >= 5.6. The latest required versions can be found at [https://reactjs.org/docs/create-a-new-react-app.html](https://reactjs.org/docs/create-a-new-react-app.html).
+
+{% include toc.md %}
+
+## Add authentication using OAuth2 and OpenID Connect (OIDC)
+
+For this demo app, we'll be using [Okta's SPA redirect model](https://developer.okta.com/docs/guides/sign-into-spa-redirect/react/main/) to authenticate and fetch user info.
+
+{% include setup/cli.md type="spa" framework="React" loginRedirectUri="http://localhost:3001/login/callback" %}
+
+## Setup the Postgresql Docker instance
+
+We'll begin by setting up a Docker instance for our Postgresql database. If you haven't already, make sure to [install Docker](https://docs.docker.com/engine/install/).
+
+1. In your root directory, create a file `docker-compose.yml`:
+```text
+services:
+  postgres:
+    container_name: "nasa-facilities"
+    image: "postgres:latest"
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_USERNAME: "${POSTGRES_USERNAME}"
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      POSTGRES_DB: "${POSTGRES_DB}"
+```
+
+2. Create a `.env` file with the following:
+```text
+POSTGRES_USERNAME=postgres
+POSTGRES_PASSWORD=postgres
+POSTGRES_DB=nasa-facilities
+```
+
+3. To use the Docker configuration we created, run `docker compose up`.
+
+4. There is a provided Postgreql data dump `./nasa-facilities_20200910.sql` in the example repo. You can import it by running the following:
```

Comment:
```suggestion
4. There is a provided PostgreSQL data dump `./nasa-facilities_20200910.sql` in the example repo. You can import it by running the following:
```

I'm thinking we should link the file here too for ease (probably the raw version)

Reply:
@alisaduncan The file is over 500 lines. I could shorten it to about 66 lines and add a note that the full dataset is included in the demo repo if you think that makes sense.

Reply:
Oh, i didn't mean to include the file in the post. I meant to add a link to the raw file within the repo.

Reply:
Ok, once the demo repo is created, I will add a link 👍